### PR TITLE
[JENKINS-73592] refresh buildhistory widget when tab was hidden before

### PR DESCRIPTION
**Pull request by mawinter69**
The current implementation of the `load` method exits early when the document is hidden to avoid unnecessary calls to the backend. But this avoids that the refreshTimeout is properly (re-)created.
This leads to 2 effects:
- when opening a job page in a new tab (ctrl+click) from a list view and then switching to the tab that the history widget is empty.
- after switching to another tab and then coming back to the job tab, the history no longer refreshes itself (See the Jira for reproduction steps)

See [JENKINS-73592](https://issues.jenkins.io/browse/JENKINS-73592).

### Testing done
Manual verified with the mentioned steps above that history widget is properly updated.

### Proposed changelog entries

- refresh buildhistory widget when tab was hidden before

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: war/src/main/js/pages/project/builds-card.js
Comment (daniel-beck) [inline war/src/main/js/pages/project/builds-card.js:33]: What the relevance of this condition?

It seems that the second problem (switching tabs) would still remain if I page through the build history first?

Is this just making this change consistent with a pre-existing but? In testing it seems I'm not getting AJAX refreshes on page 2 of the widget.

**Review by daniel-beck — APPROVED** [review]

**Comment by mawinter69** [comment]
replaced by #9624

**Pull request closed without merging**
